### PR TITLE
Add --detach=false option, to run calico-node in foreground

### PR DESCRIPTION
**Pull request by spikecurtis**
This addition allows the Calico service to be more easily integrated with init systems, which start a process and monitor it.  In this case, calicoctl is that process, and shares fate with the calico-node container.

### CODE REVIEW

File: calico_containers/calico_ctl/node.py [hunk 1]
```diff
@@ -84,14 +88,17 @@ def node(arguments):
...
                    node_image=arguments['--node-image'],
                    log_dir=arguments.get("--log-dir"),
                    ip6=arguments.get("--ip6"),
-                   as_num=arguments.get("--as"))
+                   as_num=arguments.get("--as"),
+                   detach=detach)
 
 
-def node_start(node_image, log_dir, ip="", ip6="", as_num=None):
+def node_start(node_image, log_dir, ip="", ip6="", as_num=None, detach=True):
```
Comment (tomdee) [inline calico_containers/calico_ctl/node.py]: I don't think you should be defaulting the "detach" argument here.

File: calico_containers/nose.cfg [hunk 2]
```diff
@@ -1,4 +1,4 @@
 [nosetests]
 with-coverage=1
 cover-erase=1
-cover-package=pycalico
+cover-package=pycalico,calico_ctl
```
Comment (tomdee) [inline calico_containers/nose.cfg]: @TrimBiggs FYI

**Comment by tomdee** [comment]
One minor comment. Otherwise, looks great.

File: calico_containers/calico_ctl/node.py (hunk 1 above)
Comment (spikecurtis) [inline calico_containers/calico_ctl/node.py]: probably shouldn't be defaulting any args, given how this is used.  Any objection?

**Pull request merged**